PyStemmusScope/forcing_io.py

```diff
@@ -0,0 +1,167 @@
+    to `save([-], '-ascii')` in Matlab.
+
+    Args:
+        fname (path or str): Path to the file that should be written
+        data (np.array): Array with data to write to file
+        ncols (int, optional): The number of data columns, required to correctly format
+            the ascii file when writing multiple variables.
+    """
+    matlab_fmt = ' %14.7e'
+    multi_fmt = [matlab_fmt]*ncols
+    multi_fmt[0] = ' ' + multi_fmt[0]
+    np.savetxt(fname, data, multi_fmt)
+
+
+def read_forcing_data(forcing_file):
+    """Reads the forcing data from the provided netCDF file, and applies the required
+    unit conversions before returning the read data.
+
+    Args:
+        forcing_file (str or path): Path to the netCDF file containing the forcing data
+
+    Returns:
+        dict: Dictionary containing the different variables required by STEMMUS_SCOPE
+            for the different forcing files.
+    """
+    ds_forcing = xr.open_dataset(forcing_file)
+
+    # remove the x and y coordinates from the data variables to make the numpy arrays 1D
+    ds_forcing = ds_forcing.squeeze(['x', 'y'])
+
+    data = {}
+    # Expected time format is days (as floating point) since Jan 1st 00:00.
+    data['doy_float'] = (
+        ds_forcing['time'].dt.dayofyear - 1 +
+        ds_forcing['time'].dt.hour/24 +
+        ds_forcing['time'].dt.minute/60/24
+    )
+    data['year'] = ds_forcing['time'].dt.year.astype(float)
+
+    data['t_air_celcius'] = ds_forcing['Tair'] - 273.15
+    # convert air pressure from Pa to hPa
+    data['psurf_hpa'] = ds_forcing['Psurf'] / 100
+
+    # convert CO2 concentration from 'mole fraction' to ?
+    data['co2_conv'] = ds_forcing['CO2air'] * 44 / 22.4
+
+    # convert precipitation from [kg/m2/s] to ?
+    data['precip_conv'] = ds_forcing['Precip'] / 10
+
+    data['lw_down'] = ds_forcing['LWdown']
+    data['sw_down'] = ds_forcing['SWdown']
+    data['wind_speed'] = ds_forcing['Wind']
+    data['rh'] = ds_forcing['RH']
+    data['vpd'] = ds_forcing['VPD']
+
+    data['lai'] = ds_forcing['LAI']
+    data['lai'][data['lai']<0.01] = 0.01
+
+    data['ea'] = _calculate_ea(data['t_air_celcius'], data['rh'])
+
+    return data
+
+
+def write_dat_files(data, input_dir):
+    """Fuction to write the single-data .dat files for the STEMMUS_SCOPE matlab model.
+
+    Args:
+        data (dict): Data dictionary generated by read_forcing
+        input_dir (path or str): Directory to which the different single-column .dat
+            files should be written to.
+    """
+    write_info = {
+        'doy_float': 't_.dat',
+        't_air_celcius': 'Ta_.dat',
+        'sw_down': 'Rin_.dat',
+        'lw_down': 'Rli_.dat',
+        'psurf_hpa': 'p_.dat',
+        'wind_speed': 'u_.dat',
+        'co2_conv': 'CO2_.dat',
+        'ea': 'ea_.dat',
+        'year': 'year_.dat'
+    }
+    for var, fname in write_info.items():
+        fpath = os.path.join(input_dir, fname)
```

Comment:
If we use `from pathlib import Path`, we can refactor this line as:` fpath = Path(input_dir) / fname`

Reply:
I changed all the os.path statements.